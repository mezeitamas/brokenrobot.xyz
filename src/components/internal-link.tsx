import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { Link } from 'gatsby';

type InternalLinkProps = {
    to: string;
};

const InternalLink: FunctionComponent<PropsWithChildren<InternalLinkProps>> = ({ to, children }) => {
    return (
        <div className="hover:underline">
            <Link to={to}>{children}</Link>
        </div>
    );
};

export { InternalLink };
