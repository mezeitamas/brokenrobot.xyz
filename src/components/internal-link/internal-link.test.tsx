import React from 'react';

import { create } from 'react-test-renderer';

import { InternalLink } from './internal-link';
import type { InternalLinkProps } from './internal-link';

describe('<InternalLink />', () => {
    test('should render', () => {
        // Given
        const props: InternalLinkProps = {
            to: '/about-me'
        };
        const children = 'About me';

        // When
        const component = create(<InternalLink {...props}>{children}</InternalLink>).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
