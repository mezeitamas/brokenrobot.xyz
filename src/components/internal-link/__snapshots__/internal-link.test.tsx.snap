// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InternalLink /> should render 1`] = `
<span
  className="hover:underline"
>
  <a
    href="/about-me"
  >
    About me
  </a>
</span>
`;
